src/client/camera.cpp

```diff
@@ -467,21 +467,61 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 busytime, f32 tool_r
 	 * Otherwise, default to m_cache_fov
 	 */
 
-	f32 fov_degrees;
-	PlayerFovSpec fov_spec = player->getFov();
-	if (fov_spec.fov > 0.0f) {
+	static thread_local f32 fov_degrees;
+	static thread_local bool fov_transition_active = false;
+
+	// Smooth transition variables
+	static thread_local f32 old_fov, fov_diff, transition_time;
```

Comment:
These definitely need to be class variables.

Reply:
These are only used for the following couple of lines and have a very specific purpose - I thought it'd pollute the class definition. I'll add it though.

Reply:
The problem with static variables is that they are shared between all `Camera` objects (here: on the same thread).
This will break if
1. The client ever calls the Camera from a different thread
2. There is more than one Camera

However even if none of those conditions are met, static variables like this are bad coding practice.

Reply:
Understood; thanks for the insights. :)